examples/tests/Earthfile

```diff
@@ -509,3 +510,22 @@ file-copying:
     RUN --privileged \
         --mount=type=tmpfs,target=/tmp/earthly \
         ! /usr/bin/earthly-buildkitd-wrapper.sh +test-dot-scratch
+
+run-no-cache:
+    COPY run-no-cache.earth ./Earthfile
+
+    # Run twice to allow the second one to attempt to cache things
+    RUN --privileged \
+        --entrypoint \
+        -- +test
+    RUN --privileged \
+        --entrypoint \
+        -- +test 2>&1 | perl -pe 'BEGIN {$status=0} END {exit $status} $status=1 if /\*cached\* --> .* motd2/;'  # Fail if we cached any of the motd2 lines, which are after the --no-cache
```

Comment:
Note statuses flipped here. Its easier to test that anything I dont want cached isnt, rather than everything I want is.